server/graphql/utils/requestType.ts

```diff
@@ -1,27 +1,63 @@
-import { ServerResponseInterface } from '../serverResponse'
+import { Document, Types } from 'mongoose'
+import { softDeleteRequestHelper } from './request'
 import { updateRequestGroupHelper } from './requestGroup'
 
 import { UserInputError } from 'apollo-server-errors'
 
-const updateRequestTypeHelper = (requestType, dataSources): Promise<ServerResponseInterface> => {
+import mongoose from 'mongoose'
+
+const updateRequestTypeHelper = async (requestType, dataSources): Promise<Document> => {
   if(!requestType.id) {
     throw new UserInputError('Missing argument value', { argumentName: 'id' })
   }
-  const requestGroupId = dataSources.requestTypes.getById(requestType.id.toString()).requestGroup
-  return dataSources.requestTypes.update(requestType)
-    .then(res => {
-      updateRequestGroupHelper({"id": requestGroupId}, dataSources)
-      return res
-    })
-    .catch(error => {
-      return error
-    })
+  const session = await mongoose.startSession()
+  try {
+    session.startTransaction()
+    const currentRequestType = dataSources.requestTypes.getById(requestType.id.toString())
+    const requestGroupId = currentRequestType.requestGroup.toString()
+    const res = await dataSources.requestTypes.update(requestType, session)
+    if(requestType.requestGroup) {
+      const oldRequestGroup = dataSources.requestGroups.getById(requestGroupId)
+      const newRequestGroup = dataSources.requestGroups.getById(requestType.requestGroup.toString())
+      oldRequestGroup.requestTypes = oldRequestGroup.requestTypes.filter(id => !id.equals(requestType.id))
+      newRequestGroup.requestTypes.push(requestType.id)
+      await updateRequestGroupHelper({"id": requestGroupId, "requestTypes": oldRequestGroup.requestTypes}, dataSources)
+      await updateRequestGroupHelper({"id": requestType.requestGroup.toString(), "requestTypes": newRequestGroup.requestTypes}, dataSources)
+      requestType.requestGroup = Types.ObjectId(requestType.requestGroup)
+    }
+    else {
+      await updateRequestGroupHelper({"id": requestGroupId}, dataSources)
```

Comment:
This might be what we want instead:
```
const oldRequestGroup = dataSources.requestGroups.getById(requestGroupId)
oldRequestGroup.requestTypes = oldRequestGroup.requestTypes.filter(id => !id.equals(requestType.id))
await updateRequestGroupHelper({"id": requestGroupId}, dataSources)
```